test/integration/tests/tribeRedeemer.ts

```diff
@@ -0,0 +1,228 @@
+
+    // redeem
+    await contracts.tribe.connect(signer).approve(contracts.tribeRedeemer.address, amount);
+    await contracts.tribeRedeemer.connect(signer).redeem(signer.address, amount);
+
+    const tribeBalance1 = await contracts.tribe.balanceOf(signer.address);
+    const stethBalance1 = await contracts.steth.balanceOf(signer.address);
+    const daiBalance1 = await contracts.dai.balanceOf(signer.address);
+
+    // check amount spent
+    expect(tribeBalance0.sub(tribeBalance1)).to.be.equal(amount);
+    // check amount received
+    const stethReceived = stethBalance1.sub(stethBalance0);
+    const daiReceived = daiBalance1.sub(daiBalance0);
+    expect(stethReceived).to.be.equal('0'); // got rounded down to 0, no revert
+    expect(daiReceived).to.be.equal('6'); // 6 DAI Wei
+  });
+
+  it('last redeemooor', async () => {
+    const signer = await getImpersonatedSigner(addresses.core);
+    await forceEth(signer.address);
+    const redeemerSigner = await getImpersonatedSigner(contracts.tribeRedeemer.address);
+    await forceEth(redeemerSigner.address);
+
+    const redeemChunkSizeNumber = 50_000_000;
+    const redeemChunkSize = ethers.constants.WeiPerEther.mul(redeemChunkSizeNumber);
+    const lastRedeemSize = ethers.constants.WeiPerEther.mul(10_000);
+
+    // Empty all redemption capacity, until 1000 TRIBE is left to redeem
+    await contracts.tribe.connect(signer).approve(contracts.tribeRedeemer.address, ethers.constants.MaxUint256);
+    let redeemBase = await contracts.tribeRedeemer.redeemBase();
+    while (redeemBase / 1e18 > redeemChunkSizeNumber) {
+      await contracts.tribeRedeemer.connect(signer).redeem(signer.address, redeemChunkSize);
+      await contracts.tribe.connect(redeemerSigner).transfer(signer.address, redeemChunkSize);
+      redeemBase = await contracts.tribeRedeemer.redeemBase();
+    }
+    await contracts.tribeRedeemer
+      .connect(signer)
+      .redeem(signer.address, (await contracts.tribeRedeemer.redeemBase()).sub(lastRedeemSize));
+
+    const tribeBalance0 = await contracts.tribe.balanceOf(signer.address);
+    const stethBalance0 = await contracts.steth.balanceOf(signer.address);
+    const daiBalance0 = await contracts.dai.balanceOf(signer.address);
+
+    // last redeem
+    await contracts.tribeRedeemer.connect(signer).redeem(signer.address, lastRedeemSize);
+
+    const tribeBalance1 = await contracts.tribe.balanceOf(signer.address);
+    const stethBalance1 = await contracts.steth.balanceOf(signer.address);
+    const daiBalance1 = await contracts.dai.balanceOf(signer.address);
+
+    // check amount spent
+    expect(tribeBalance0.sub(tribeBalance1)).to.be.equal(lastRedeemSize);
+    // check amount received
+    const stethReceived = stethBalance1.sub(stethBalance0);
+    const daiReceived = daiBalance1.sub(daiBalance0);
+    expect(stethReceived).to.be.at.least(ethers.constants.WeiPerEther.mul(109).div(100)); // [1.09, 1.12] stETH
```

Comment:
Would be good to do `n` claims of 2 or 3 different sizes before making some final claim and checking all the values are consistent

Reply:
ok